src/request-pipeline/context/index.ts

```diff
@@ -426,6 +426,15 @@ export default class RequestPipelineContext extends BaseRequestPipelineContext {
         this.goToNextStage = false;
     }
 
+    closeConnectionOnError (): void {
+        this.req.destroy();
+
+        // For Node versions greater than 15.4, we check the socket. If req.destroy didn't destroy it, we call the corresponding method of the socket.
+        if (!this.req.socket?.destroyed)
+            this.req.socket.destroy();
```

Comment:
What if `this.req.socket` expression returns `null` or `undefined`? It looks like we will get null ref exception